### PR TITLE
Add codecov

### LINKED ISSUE #226: Introducing Code Coverage
I'd like to consider implementing code coverage, referring to the following repository.
The only thing we need to consider is that we can probably only do it nightly.
https://github.com/containers/oci-spec-rs

### PR DESCRIPTION
**Pull request by chenyukang**
Fix #226

### CODE REVIEW

**Comment by chenyukang** [comment]
@utam0k Please merge it and update the badge. https://app.codecov.io/gh/containers/youki/settings
I don't have the permissions.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -54,18 +54,33 @@ jobs:
...
+          RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
       - name: Run doc tests
         run: cargo test --doc
       - name: Run cgroup tests
         working-directory: cgroups
         run: cargo test
+        env:
+          CARGO_INCREMENTAL: '0'
+          RUSTFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
+          RUSTDOCFLAGS: '-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zpanic_abort_tests'
```
Comment (utam0k) [inline .github/workflows/main.yml:77]: Why does the cargo test run twice?

Comment (chenyukang) [reply .github/workflows/main.yml:77]: > Why does the cargo test run twice?

It's in different directories, one in top-level and one in cgroups, I didn't change it.

Comment (utam0k) [reply .github/workflows/main.yml:77]: Oops, wrong working directory. I overlooked it.

**Review by utam0k — APPROVED** [review]
lgtm

**Pull request merged**